DigitalLearningSolutions.Web.Tests/Controllers/Login/LoginControllerTests.cs

```diff
@@ -92,6 +95,65 @@ public void Successful_sign_in_should_render_home_page()
                 .WithControllerName("Home").WithActionName("Index");
         }
 
+        [Test]
+        public void Successful_sign_in_with_local_return_url_should_redirect_to_return_url()
+        {
+            //Given
+            var expectedAdmin = UserTestHelper.GetDefaultAdminUser();
+            var expectedDelegates = new List<DelegateUser> { UserTestHelper.GetDefaultDelegateUser() };
+            A.CallTo(() => userService.GetUsersByUsername(A<string>._))
+                .Returns((expectedAdmin, expectedDelegates));
+            A.CallTo(() => loginService.VerifyUsers(A<string>._, A<AdminUser>._, A<List<DelegateUser>>._))
+                .Returns((expectedAdmin, expectedDelegates));
+            A.CallTo(() => userService.GetUsersWithActiveCentres(A<AdminUser>._, A<List<DelegateUser>>._))
+                .Returns((expectedAdmin, expectedDelegates));
+            A.CallTo(() => userService.GetUserCentres(A<AdminUser>._, A<List<DelegateUser>>._))
+                .Returns(
+                    new List<CentreUserDetails> { new CentreUserDetails(1, "Centre 1", true, true) });
+
```

Comment:
I wonder if we could separate this into a private method `GivenSignInIsSuccessful` to make these two tests a bit easier to read.